api/data/seattle_db.go

```diff
@@ -83,58 +88,139 @@ func (c *Client) SeattleOfficerMetadata() *DepartmentMetadata {
 				Path:        "/seattle/officer/search",
 				QueryParams: []string{"first_name", "last_name"},
 			},
+			"historical-exact": {
+				Path:        "/seattle/officer/historical",
+				QueryParams: []string{"badge", "first_name", "last_name"},
+			},
 		},
 	}
 }
 
-// SeattleGetOfficerByBadge invokes seattle_get_officer_by_badge_p
-func (c *Client) SeattleGetOfficerByBadge(badge string) (*SeattleOfficer, error) {
-	ofc := SeattleOfficer{}
-	err := c.pool.QueryRow(context.Background(),
+// SeattleGetOfficerByBadge returns an officer by their badge. It searches the full historical
+// roster list but only returns the most recent entry.
+func (c *Client) SeattleGetOfficerByBadge(badge string) ([]*SeattleOfficer, error) {
+	rows, err := c.pool.Query(context.Background(),
+		`
+			WITH max_roster AS (SELECT MAX(date) max_date FROM seattle_officers)
+			SELECT
+				o.date,
+				o.badge,
+				o.full_name,
+				o.first_name,
+				o.middle_name,
+				o.last_name,
+				o.title,
+				o.unit,
+				o.unit_description,
+				CASE WHEN o.date = m.max_date THEN TRUE ELSE FALSE END is_current
+			FROM seattle_officers o
+			CROSS JOIN max_roster m
+			WHERE o.badge = $1
+			ORDER BY o.date DESC
+			LIMIT 1;
+		`,
+		badge,
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	return seattleMarshalOfficerRows(rows)
+}
+
+// SeattleGetOfficerByBadgeHistorical returns an officer by their badge. It searches 
+// the full historical roster list and returns all entries in descending date order.
+func (c *Client) SeattleGetOfficerByBadgeHistorical(badge string) ([]*SeattleOfficer, error) {
+	rows, err := c.pool.Query(context.Background(),
 		`
+			WITH max_roster AS (SELECT MAX(date) max_date FROM seattle_officers)
 			SELECT
-				date,
-				badge,
-				full_name,
-				first_name,
-				middle_name,
-				last_name,
-				title,
-				unit,
-				unit_description
-			FROM seattle_get_officer_by_badge_p (badge := $1);
+				o.date,
+				o.badge,
+				o.full_name,
+				o.first_name,
+				o.middle_name,
+				o.last_name,
+				o.title,
+				o.unit,
+				o.unit_description,
+				CASE WHEN o.date = m.max_date THEN TRUE ELSE FALSE END is_current
+			FROM seattle_officers o
+			CROSS JOIN max_roster m
+			WHERE o.badge = $1
+			ORDER BY o.date DESC;
```

Comment:
I had originally done this hoping that I could reuse it for all the SQL queries but I was only able to put something together for the badge query, so I decided to just leave it separate. It felt like it was added complexity for not much value. It may be something we can do later, though.